src/Execution/ModelsLoader/CountModelsLoader.php

```diff
@@ -32,7 +32,7 @@ public static function extractCount(Model $model, string $relationName): int
          */
         $countAttributeName = Str::snake("{$relationName}_count");
 
-        $count = $model->getAttribute($countAttributeName);
+        $count = (int) $model->getAttribute($countAttributeName);
         assert(is_int($count), 'avoid runtime check in production since the return type validates this anyway');
 
         return $count;
```

Comment:
```suggestion
        $count = $model->getAttribute($countAttributeName);
        if (! is_numeric($count)) {
            $nonNumericCount = Utils::printSafe($count);
            throw new \Exception('Expected numeric count, got: {$nonNumericCount}.");
        }

        return (int) $count;
```

I am always wary of unchecked casts, since PHP will do the most senseless cast and just give a wrong result instead of throwing.